content/articles/using-django-in-kivy-application/index.md

```diff
@@ -0,0 +1,596 @@
+### Using Django in Kivy applications.
+
+### Introduction.
+
+Kivy is a popular library for developing applications that can run across all major platforms.
+There may, however, be situations that require your application to access and manipulate data being provided by a web server 
+hosted at a site.In this tutorial, we are going to learn how to make requests to a server
+using the `requests` library. We are going to use Django to develop the server that holds the data we want to manipulate.
+
+### Prerequisites.
+For you to follow along, it's important that:
+1. You have `Django` and `djangorestframework` installed.
+2. You are familiar with building APIs using Django REST framework.
+3. You have the `kivy` library installed. You can run `pip install kivy`.
+4. Basic knowledge of python is important, especially the concept of object-oriented programming in python.
+
+### Takeaways.
+1. Improve your python skills. 
+2. Learn how to build applications using the `kivy` library.
+3. Learn how to use Django for your kivy applications.
+
+### Getting Started.
+We are going to create a simple todo application that allows one to view available tasks and
+offers an option for adding a new task.We will begin by creating the tasks API with `djangorestframework`
+and then create our application using `kivy`. We will use the `requests` library to make 
+requests to our Django server.
+
+### Creating the tasks API.
+In a folder of choice, let's create a new project by running `django-admin startproject TodoAPI`.Cd into the `TodoAPI` project and create a new app 
+`tasks` that will handle creation of tasks. Run `python3 manage.py startapp tasks`. Your project structure should look something like this:
+
+```buildoutcfg
+.
+└── TodoAPI
+    ├── manage.py
+    ├── tasks
+    │   ├── admin.py
+    │   ├── apps.py
+    │   ├── __init__.py
+    │   ├── migrations
+    │   │   └── __init__.py
+    │   ├── models.py
+    │   ├── tests.py
+    │   └── views.py
+    └── TodoAPI
+        ├── asgi.py
+        ├── __init__.py
+        ├── settings.py
+        ├── urls.py
+        └── wsgi.py
+
+
+
+```
+Edit the `settings.py` file as follows under `INSTALLED APPS`:
+`TodoAPI/settings.py`
+
+```python
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'tasks',
+    'rest_framework'
+
+```
+Let's proceed and create our tasks model.Edit the `models.py` file to look as follows:
+
+`tasks/models.py`
+```python
+from django.db import models
+
+# Create your models here.
+class Task(models.Model):
+    name = models.CharField(max_length=200)
+    def __str__(self):
+        return self.name
+```
+
+Let's create a `serializer.py` file that handles serialization and deserialization of task instances.Add the following lines of code to it:
```

Comment:
`serialization and deserialization`

Explain in 1 - 2 lines about these, and reference links for these